pkg/apis/extensions/types.go

```diff
@@ -268,6 +268,11 @@ const (
 
 	// Replace the old RCs by new one using rolling update i.e gradually scale down the old RCs and scale up the new one.
 	RollingUpdateDeploymentStrategyType DeploymentStrategyType = "RollingUpdate"
+
+	// The Deployment controller is not going to orchestrate ReplicaSets for
+	// Deployments that use this strategy. Useful for delegating orchestration of
+	// certain Deployments to external controllers.
+	NoopDeploymentStrategyType DeploymentStrategyType = "Noop"
```

Comment:
Yeah, a concrete example of an external controller to which strategy is delegated would be great. 

See also my comment here https://github.com/kubernetes/kubernetes/issues/39337#issuecomment-270324701 but probably best to discuss that in the issue rather than the PR.